### ISSUE TITLE
Dealing with breaking changes

### ISSUE THREAD

**Issue by michael** [michael/hub#45]
We need a better way to coordinate breaking changes on the API side. At least for me it is quite hard to keep up with a number of API changes that need to be reflected on the composer side.

My suggestion: Let's try to apply API changes in smaller increments, so we can nail them one by one.

Proposed workflow: 
1. Suggest an API change using a GitHub Issue. E.g. `POST /register` -> `POST /users`
2. Coordinate implementation, so the change is reflected in both repos (hub, composer)
3. Close Issue

Thoughts?

**Comment by vectorsize** [comment]
Totally agree – this sort of things can block everyone else's work…

**Comment by michael** [comment]
Once I'm ready adapting the new API I'm going to host a bleeding edge hub on the substance server.

Will be available at: `http://edge.substance.io` or `http://substance.io:3001`

Maybe we should also look into some continuous integration solutions. Something like this:
- Every push to hub:master triggers a deploy to edge.substance.io
- However the deploy only happens if the testsuite passes through
- If there are errors we should be able to easily check the log on a webinterface (e.g. http://edge.substance.io/status)

Much like of Github's "battlestation fully operational".

Will need some time to set this up. But I guess this could be incredibly helpful for speeding up development.

**Comment by yuchi** [comment]
About APIs changes: totally agree. I'm sorry for what happened in the last refactor. The testsuite is also there to consolidate development.

About CI: that's perfect. I'm a little bit worried about the database. Is it going to be a different one?

**Comment by michael** [comment]
Yeah it's a different database for our staging environment.

I just got the new server instance live:

See: http://edge.substance.io (you can also access the admin panel -> http://edge.substance.io/admin)

**Comment by michael** [comment]
@yuchi I like how the admin panel looks. Now I see why bootstrap is a good idea here. :)

**Comment by yuchi** [comment]
I hate Bootstrap... but god it's so fast to work with :\

**Comment by vectorsize** [comment]
yeah @yuchi but any of the like are the same, just keeps you up and running quick… I had bookmarked other possible alternatives.
[edit] removed the links, these where not the ones haha
